source/gui/settingsDialogs.py

```diff
@@ -5961,6 +5962,242 @@ def onSave(self):
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			# Default to 2.0x if module not available or value not found
+			log.debug(f"Error getting current zoom: {e}")
+			try:
+				defaultIndex = self.zoomValues.index(2.0)
+				self.defaultZoomList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultZoomList.SetSelection(0)
+
+		# FILTER
+		try:
+			# Get current filter from magnifier module
+			currentFilter = magnifier.getCurrentFilter()
+
+			# Ensure it's a string
+			currentFilter = str(currentFilter)
+
+			if currentFilter in self.defaultFilterList.GetStrings():
+				self.defaultFilterList.SetSelection(self.defaultFilterList.GetStrings().index(currentFilter))
+			else:
+				self.defaultFilterList.SetSelection(0)
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			log.debug(f"Error getting current filter: {e}")
+			try:
+				defaultIndex = self.defaultFilterList.GetStrings().index("None")
+				self.defaultFilterList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultFilterList.SetSelection(0)
+
+		# FULLSCREEN MODE
+		try:
+			currentFullscreenMode = magnifier.getCurrentFullscreenMode()
+
+			# Ensure it's a string
+			currentFullscreenMode = str(currentFullscreenMode)
+
+			if currentFullscreenMode in self.defaultFullscreenModeList.GetStrings():
+				self.defaultFullscreenModeList.SetSelection(
+					self.defaultFullscreenModeList.GetStrings().index(currentFullscreenMode)
+				)
+			else:
+				self.defaultFullscreenModeList.SetSelection(0)
+
+		except (ImportError, ValueError, AttributeError, IndexError) as e:
+			log.debug(f"Error getting current fullscreen mode: {e}")
+			try:
+				defaultIndex = self.defaultFullscreenModeList.GetStrings().index("center")
+				self.defaultFullscreenModeList.SetSelection(defaultIndex)
+			except ValueError:
+				self.defaultFullscreenModeList.SetSelection(0)
+
+		# KEEP MOUSE CENTERED
+		try:
+			keepMouseCentered = config.conf["magnifier"]["keepMouseCentered"]
+			# Convert string to boolean if necessary
+			if isinstance(keepMouseCentered, str):
+				keepMouseCentered = keepMouseCentered.lower() in ("true", "1", "yes")
+			self.keepMouseCenteredCheckBox.SetValue(bool(keepMouseCentered))
+		except (KeyError, AttributeError):
+			self.keepMouseCenteredCheckBox.SetValue(False)
+
+		# SAVE SHORTCUT CHANGES
+		try:
+			saveShortcutChanges = config.conf["magnifier"]["saveShortcutChanges"]
+			# Convert string to boolean if necessary
+			if isinstance(saveShortcutChanges, str):
+				saveShortcutChanges = saveShortcutChanges.lower() in ("true", "1", "yes")
+			self.saveShortcutChangesCheckBox.SetValue(bool(saveShortcutChanges))
+		except (KeyError, AttributeError):
+			self.saveShortcutChangesCheckBox.SetValue(False)
+
+	def onPanelActivated(self):
+		"""Called when the panel is activated/shown."""
+		super().onPanelActivated()
+		# Refresh the selection when panel is shown
+		self._updateCurrentSelection()
+
+	def onSave(self):
```

Comment:
this functions looks very different to other `onSave` functions. Can you explain why the differences are necessary?